tools/accuracy_checker/accuracy_checker/annotation_converters/criteo_kaggle_dac.py

```diff
@@ -95,36 +143,34 @@ def convert(self, check_content=False, **kwargs):
         subfolder = 0
         filecnt = 0
 
-        data = np.load(self.src)
-        x_int = data['X_int']
-        x_cat = data['X_cat']
-        y = data['y']
-        samples, cat_feat = x_cat.shape
-
-        samples = (samples // self.batch) * self.batch
+        self.load_data_file()
+        samples = (self.count // self.batch) * self.batch
         start = 0
 
         if self.subsample:
-            samples = self.subsample * self.batch if samples > self.subsample * self.batch else samples
+            samples = self.subsample if samples > self.subsample else samples
         elif self.validation:
             start = samples // 2
 
-        for i in range(start, samples - self.batch + 1, self.batch):
+        for i in range(start, samples):
             c_input = input_folder / "{:02d}".format(subfolder)
             c_input = c_input / "{:06d}.npz".format(i)
 
             if self.save_preprocessed_features:
+                x_int, x_cat, y = self.get_data(i)
                 if not c_input.parent.exists():
                     c_input.parent.mkdir(parents=True)
 
                 sample = {
-                    self.dense_features: np.log1p(x_int[i:i+self.batch, ...]),
+                    self.dense_features: np.log1p(x_int),
                     self.lso_features: np.dot(np.expand_dims(np.linspace(0, self.batch - 1, num=self.batch), -1),
-                                              np.ones((1, cat_feat))).T
+                                              np.ones((1, self.cat_feat))).T
                 }
 
                 for name in self.sparse_features.keys():
-                    sample[name] = x_cat[i:i+self.batch, self.sparse_features[name]].T
+                    x_cat_batch = x_cat[:, self.sparse_features[name]]
+                    x_cat_batch = x_cat_batch % self.max_ind_range if self.max_ind_range > 0 else x_cat_batch
```

Comment:
not clear logic for me, better to use None as identifier that max_ind_range is not set instead of 0.
0 gives user opprotunity to set this value manually even it does not required